Housekeeping fix brackets

LGM-Doyle opened the pull request:
This PR fixes all of the warnings exposed by -Wparentheses by adding the requested brackets.

LGM-Doyle committed:
housekeeping:  Fix warnings from -Wparentheses.

LGM-Doyle committed:
housekeeping:  Fix warnings from -Wparentheses in parsers.

The warnings from -Wparentheses are spurious because the '>' operator is
a boost::phoenix parsing operator not a comparison operator.

Adding brackets silences the warning.

adrianbroher reviewed (commented):
The change itself looks good, but I don't know if this should be merged or if the nagging compiler should be silenced.

geoffthemedio reviewed (changes requested):
Adding brackets to silence warnings is fine, but please don't change the indenting when doing so

geoffthemedio reviewed (changes requested):
parser indenting problems, incomplete list

LGM-Doyle committed:
Fix indentation for Review #1043.

LGM-Doyle committed:
Fix logic error in Condition.cpp from review #1043.

Fix three cases where condition logic was:

local-invariant && root-candidate || root-candidate-invariant

,but should be:

(local-invariant) && (root-candidate || root-candidate-invariant)
